packages/mobile/src/verify/reducer.ts

```diff
@@ -0,0 +1,340 @@
+  | Error
+
+export interface KomenciContext {
+  errorTimestamps: number[]
+  unverifiedMtwAddress: string | null
+  sessionActive: boolean
+  sessionToken: string
+  callbackUrl: string | undefined
+  captchaToken: string
+}
+
+export interface State {
+  status: AttestationsStatus & { komenci: boolean }
+  actionableAttestations: ActionableAttestation[]
+  currentState: InternalState
+  komenci: KomenciContext
+  komenciAvailable: boolean | undefined
+  phoneHash?: string
+  e164Number?: string
+  retries: number
+  TEMPORAR_override_withoutVerification?: boolean
+}
+
+const initialState: State = {
+  komenci: {
+    errorTimestamps: [],
+    unverifiedMtwAddress: null,
+    sessionActive: false,
+    sessionToken: '',
+    callbackUrl: undefined,
+    captchaToken: '',
+  },
+  status: {
+    isVerified: false,
+    numAttestationsRemaining: 3,
+    total: 0,
+    completed: 0,
+    komenci: true,
+  },
+  actionableAttestations: [],
+  retries: 0,
+  currentState: idle(),
+  komenciAvailable: undefined,
+  TEMPORAR_override_withoutVerification: undefined,
+}
+
+export const reducer = createReducer(initialState, (builder) => {
+  builder
+    .addCase(rehydrate, (state, action) => {
+      // hack to allow rehydrate actions here
+      const hydrated = getRehydratePayload((action as unknown) as RehydrateAction, 'verify')
+      return {
+        ...state,
+        ...hydrated,
+        komenci: {
+          ...state.komenci,
+          ...hydrated.komenci,
+          captchaToken: initialState.komenci.captchaToken,
+        },
+        retries: 0,
+        currentState: idle(),
+      }
+    })
+    .addCase(stop, (state) => {
+      return { ...state, currentState: idle() }
+    })
+    .addCase(start, (state, action) => {
+      return {
+        ...state,
+        e164Number: action.payload.e164Number,
+        currentState: preparing(),
+      }
+    })
+    .addCase(ensureRealHumanUser, (state) => {
+      return {
+        ...state,
+        currentState: ensuringRealHumanUser(),
+      }
+    })
+    .addCase(startKomenciSession, (state) => {
+      return {
+        ...state,
+        currentState: startingKomenciSession(),
+      }
+    })
+    .addCase(fetchPhoneNumberDetails, (state) => {
+      return {
+        ...state,
+        currentState: fetchingPhoneNumberDetails(),
+      }
+    })
+    .addCase(setUseKomenci, (state, action) => {
+      return {
+        ...state,
+        shouldUseKomenci: action.payload,
+        komenci: initialState.komenci,
+      }
+    })
+    .addCase(setPhoneHash, (state, action) => {
+      return {
+        ...state,
+        phoneHash: action.payload,
+      }
+    })
+    .addCase(fetchMtw, (state) => {
+      return {
+        ...state,
+        currentState: fetchingMtw(),
+      }
+    })
+    .addCase(fetchOnChainData, (state) => {
+      return {
+        ...state,
+        currentState: fetchingOnChainData(),
+      }
+    })
+    .addCase(setVerificationStatus, (state, action) => {
```

Comment:
Does this logic here handle the case where the user backgrounds Valora while waiting for the tx to be confirmed?
https://app.zenhub.com/workspaces/wallet-backlog-board-60092d006a84e3001d0d5936/issues/celo-org/celo-monorepo/6473

Reply:
Essentially ATM this PR covers only the part before the actual verification process starts. Meaning that actual request and reveal of attestations is still happening in `doVerificationFlowSaga`.